### ISSUE TITLE
[stable/elasticsearch]: Can I get this chart to work with elasticsearch version 5.5.3?

### ISSUE THREAD

**Issue by graziergeek** [helm/charts#8640]
**Is this a request for help?**:
Yes
---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):

Neither.  It's a request for information.

**Version of Helm and Kubernetes**:

Helm Client only version 2.10.0
Kubernetes 1.9.7-gke.6

**Which chart**:

stable/elasticsearch

**What happened**:

I've been trying to get Elasticsearch running on K8s using this newly-promoted-to-stable helm chart, which works fine, BTW, for elasticsearch v 6.4.2. However, we're tied to a grails app that requires elasticsearch v 5.5.3, which we don't have the ability to upgrade. I've downgraded the elasticsearch image version in the chart to 5.5.3 (and also tried v 5.6.12) but it fails to start.

Client pods fail with inability to connect to `elasticsearch-discovery` headless service.

master-0 (the only master that attempts to start) fails with:
```
1) Error injecting constructor, ElasticsearchException[java.io.IOException: failed to read [id:15, legacy:false, file:/usr/share/elasticsearch/data/nodes/0/_state/global-15.st]]; nested: IOException[failed to read [id:15, legacy:false, file:/usr/share/elasticsearch/data/nodes/0/_state/global-15.st]]; nested: ElasticsearchException[Unknown license version found, please upgrade all nodes to the latest elasticsearch-license plugin];
  at org.elasticsearch.gateway.GatewayMetaState.<init>(Unknown Source)
  while locating org.elasticsearch.gateway.GatewayMetaState
    for parameter 4 at org.elasticsearch.gateway.GatewayService.<init>(Unknown Source)
  while locating org.elasticsearch.gateway.GatewayService
Caused by: ElasticsearchException[java.io.IOException: failed to read [id:15, legacy:false, file:/usr/share/elasticsearch/data/nodes/0/_state/global-15.st]]; nested: IOException[failed to read [id:15, legacy:false, file:/usr/share/elasticsearch/data/nodes/0/_state/global-15.st]]; nested: ElasticsearchException[Unknown license version found, please upgrade all nodes to the latest elasticsearch-license plugin];
```

My question is, is there a way to make this chart work with v 5.5.3 and, if so, can someone outline the necessary steps?

Many thanks! 

**What you expected to happen**:

This is not unexpected.

**How to reproduce it** (as minimally and precisely as possible):

1. `helm template <this-char> ...` with all defaults.
1. Change all elasticsearch image versions from `6.4.2` to `5.5.3` (note I used the std elasticsearch docker image: `docker.elastic.co/elasticsearch/elasticsearch:6.4.2`)
1. `kubectl apply --recursive -f elasticsearch/templates/`

**Anything else we need to know**:

Any help is greatly appreciated and I know it's a long shot. Thanks!

**Comment by RuriRyan** [comment]
Any news on this one?

**Comment by graziergeek** [comment]
We ended up using a variation of the "multi-node" solution found in [this Medium post](https://medium.com/google-cloud/a-guide-to-deploy-elasticsearch-cluster-on-google-kubernetes-engine-52f67743ee98) -- no SSD and disabled boot checking.

This is not the high availability provided by this chart but we needed to keep an old app running for a while longer before we can upgrade ES.  I do not support this solution any farther than just getting you by.  Clearly, the way forward for HA ES is to use this chart or the ES Operator.

I will close this issue.
